pddl/MultiAgentPDDL.g4

```diff
@@ -0,0 +1,595 @@
+    ;
+
+/************* EXPRESSIONS ****************************/
+
+fExp
+    : NUMBER
+    | '(' binaryOp fExp fExp2 ')'
+    | '(' '-' fExp ')'
+    | fHead
+    ;
+
+// This is purely a workaround for an ANTLR bug in tree construction
+// http://www.antlr.org/wiki/display/ANTLR3/multiple+occurences+of+a+token+mix+up+the+list+management+in+tree+rewrites
+fExp2
+    : fExp
+    ;
+
+fHead
+    : '(' functionSymbol term* ')'
+    | functionSymbol
+    ;
+
+effect
+    : '(' 'and' cEffect* ')'
+    | cEffect
+    ;
+
+cEffect
+    : '(' 'forall' '(' typedVariableList ')' effect ')'
+    | '(' 'when' goalDesc condEffect ')'
+    | pEffect
+    ;
+
+pEffect
+    : '(' assignOp fHead fExp ')'
+    | '(' 'not' atomicTermFormula ')'
+    | atomicTermFormula
+    ;
+
+// TODO: why is this different from the "and cEffect" above? Does it matter?
+condEffect
+    : '(' 'and' pEffect* ')'
+    | pEffect
+    ;
+
+// TODO: should these be uppercase & lexer section?
+binaryOp
+    : '*'
+    | '+'
+    | '-'
+    | '/'
+    ;
+
+binaryComp
+    : '>'
+    | '<'
+    | '='
+    | '>='
+    | '<='
+    ;
+
+assignOp
+    : 'assign'
+    | 'scale-up'
+    | 'scale-down'
+    | 'increase'
+    | 'decrease'
+    ;
+
+/************* DURATIONS  ****************************/
+
+durationConstraint
+    : '(' 'and' simpleDurationConstraint+ ')'
+    | '(' ')'
+    | simpleDurationConstraint
+    ;
+
+simpleDurationConstraint
+    : '(' durOp '?duration' durValue ')'
+    | '(' 'at' timeSpecifier simpleDurationConstraint ')'
+    ;
+
+durOp
+    : '<='
+    | '>='
+    | '='
+    ;
+
+durValue
+    : NUMBER
+    | fExp
+    ;
+
+daEffect
+    : '(' 'and' daEffect* ')'
+    | timedEffect
+    | '(' 'forall' '(' typedVariableList ')' daEffect ')'
+    | '(' 'when' daGD timedEffect ')'
+    | '(' assignOp fHead fExpDA ')'
+    ;
+
+timedEffect
+    : '(' 'at' timeSpecifier cEffect ')'
+    | '(' 'at' timeSpecifier fAssignDA ')'
+    | '(' assignOpT fHead fExp ')' /* fExpT in BNF but not making sense */
+    ;
+
+fAssignDA
+    : '(' assignOp fHead fExpDA ')'
+    ;
+
+fExpDA
+    : '(' ((binaryOp fExpDA fExpDA) | ('-' fExpDA)) ')'
+    | '?duration'
+    | fExp
+    ;
+
+assignOpT
+    : 'increase'
+    | 'decrease'
+    ;
+
+/*
+ * We should have fExpT according to:
+ * http://www.plg.inf.uc3m.es/ipc2011-deterministic/attachments/OtherContributions/kovacs-pddl-3.1-2011.pdf
+
+fExpT
+	: '(*' fExp #t ')'
+	| '(*' #t fExp ')'
+	| #t
+	;
+ */
+
+/************* PROBLEMS ****************************/
+
+problem
+    : '(' 'define' problemDecl problemDomain requireDef? objectDecl? init goal probConstraints? metricSpec?
+    // lengthSpec? This is not defined anywhere in the BNF spec
+    ')'
+    ;
+
+problemDecl
+    : '(' 'problem' name ')'
+    ;
+
+problemDomain
+    : '(' ':domain' name ')'
+    ;
+
+objectDecl
+    : '(' ':objects' objectList? ')'
+    ;
+
+objectList
+    : objectDeclaration+
+    ;
+
+objectDeclaration
+    : objectName '-' objectType
```

Comment:
Please change to `objectName+`. See http://www.plg.inf.uc3m.es/ipc2011-deterministic/attachments/Resources/kovacs-pddl-3.1-2011.pdf , specifically the rule `<function typed list (x)> ::= x+ - <function type> <function typed list(x)>`. Notice the "+" for "x", where it is used in rule `<object declaration> ::= (:objects <typed list (name)>)`. Some of the tests don't parse without this.

Reply:
A single character escaped me - apologies. Resolved!